src/Misc/AresData.cpp

```diff
@@ -0,0 +1,109 @@
+#include "AresData.h"
+
+#include <ASMMacros.h>
+#include <Phobos.h>
+#include <Utilities/Debug.h>
+#include <CRC.h>
+
+#include <tlhelp32.h>
+
+class TechnoClass;
+class TechnoTypeClass;
+
+uintptr_t AresData::PhobosBaseAddress = 0x0;
+uintptr_t AresData::AresBaseAddress = 0x0;
+HMODULE AresData::AresDllHmodule = nullptr;
+int AresData::AresVersionId = AresData::Version::Unknown;
+bool AresData::CanUseAres = false;
+DWORD AresData::AresFunctionOffsetsFinal[AresData::AresFunctionCount];
+
+#ifndef PHOBOS_DLL
+#define PHOBOS_DLL "Phobos.dll"
+#endif
+
+void AresData::GetGameModulesBaseAddresses()
+{
+	HANDLE hCurrentProcess = GetCurrentProcess();
+
+	HANDLE hSnap = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE | TH32CS_SNAPMODULE32, GetProcessId(hCurrentProcess));
+	if (hSnap != INVALID_HANDLE_VALUE)
+	{
+		MODULEENTRY32 modEntry{ };
+		modEntry.dwSize = sizeof(modEntry);
+		if (Module32First(hSnap, &modEntry))
+		{
+			do
+			{
+				Debug::LogDeferred("Module %s base address : 0x%p.\n", modEntry.szModule, modEntry.modBaseAddr);
+				if (!_strcmpi(modEntry.szModule, "Ares.dll"))
+				{
+					AresData::AresBaseAddress = (uintptr_t)modEntry.modBaseAddr;
+				}
+				else if(!_strcmpi(modEntry.szModule, PHOBOS_DLL))
+				{
+					AresData::PhobosBaseAddress = (uintptr_t)modEntry.modBaseAddr;
+				}
```

Comment:
```suggestion
				if (!_strcmpi(modEntry.szModule, "Ares.dll"))
					AresData::AresBaseAddress = (uintptr_t)modEntry.modBaseAddr;
				else if(!_strcmpi(modEntry.szModule, PHOBOS_DLL))
					AresData::PhobosBaseAddress = (uintptr_t)modEntry.modBaseAddr;
```

Does this use the filename which it was compiled with or the actual filename of the DLL?

Reply:
Compiled with ($(ProjectName)), cuz I didn't know how to get the filename at runtime

Reply:
No I mean if it's uses the internal name (`Original filename` in metadata) then it's fine, otherwise what if someone renames it?

Reply:
Je ne sais pas.
I also tried to print all submodules names in log file within cncnet5.dll or DP at later game, but Phobos wasn't listed. I don't know why

Reply:
Modified to use `Original filename`.
Still, I wouldn't ever advise renaming Ares.dll and Phobos.dll, as Syringe loads hooks in alphabetical order and having APhobos.dll and ZAres.dll (or the other way around) will mess things up.